### PR TITLE
[PC-9844] upgrade RN and activate hermes on iOS

### PR DESCRIPTION
**Pull request by marionnousvalentin**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-9844
## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -6,15 +6,17 @@ platform :ios, '10.0'
...
+     :hermes_enabled => true
+   )
   # Third party deps podspec link, to include if 0.60 autolinking not supported
 
   # Enables Flipper.
   #
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable these next few lines.
-  use_flipper!({ 'Flipper' => '0.92.0' })
+  use_flipper!({ 'Flipper' => '0.92.0' , 'Flipper-Folly' => '~> 2.5'})
```
Comment (antoinewg) [inline ios/Podfile:19]: Ca build pas sinon ?

File: metro.config.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ module.exports = {
     getTransformOptions: async () => ({
       transform: {
         experimentalImportSupport: false,
-        inlineRequires: false,
+        inlineRequires: true,
```
Comment (antoinewg) [inline metro.config.js:82]: C'est pour quoi ça ?

File: package.json [hunk 3]
```diff
@@ -95,9 +95,9 @@
     "lottie-react-native": "^4.0.0",
     "make-plural": "^6.2.2",
     "patch-package": "^6.2.0",
-    "react": "16.13.1",
+    "react": "17.0.1",
     "react-error-boundary": "^3.1.0",
-    "react-native": "0.63.3",
+    "react-native": "0.64.2",
```
Comment (antoinewg) [inline package.json:100]: 🎉

**Review by antoinewg — APPROVED** [review]
Top. Juste 2 questions mais approuvée

File: ios/Podfile (hunk 1 above)
Comment (marionnousvalentin) [reply ios/Podfile:19]: yes

File: metro.config.js (hunk 2 above)
Comment (marionnousvalentin) [reply metro.config.js:82]: c'est préisé dans la doc, j'avoue que je n'ai pas creusé

**Pull request merged**